src/comments.js

```diff
@@ -193,12 +193,12 @@ function attach(comments, ast, text) {
           }
         }
         tiesToBreak.push(comment);
-      }
-      else if(precedingNode) {
+      } else if(precedingNode) {
         addTrailingComment(precedingNode, comment);
-      }
-      else if(followingNode) {
+      } else if(followingNode) {
         addLeadingComment(followingNode, comment);
+      } else if (enclosingNode) {
+        addDanglingComment(enclosingNode, comment);
       }
```

Comment:
I tried to add an else case here and throw, nothing triggered it in the test suite.